src/nodes/chunk_append/exec.c

```diff
@@ -925,16 +927,42 @@ chunk_append_get_lock_pointer()
 static List *
 constify_restrictinfos(PlannerInfo *root, List *restrictinfos)
 {
+	List *additional_list = NIL;
+
 	ListCell *lc;
 
 	foreach (lc, restrictinfos)
 	{
 		RestrictInfo *rinfo = lfirst(lc);
+		Expr *constified = (Expr *) estimate_expression_value(root, (Node *) rinfo->clause);
 
-		rinfo->clause = (Expr *) estimate_expression_value(root, (Node *) rinfo->clause);
+		if (!equal(rinfo->clause, constified))
+		{
+			/*
+			 * We have constified something, so try applying the time_bucket
+			 * transformations again. This might allow us to exclude chunks
+			 * based on a parameterized time_bucket expression.
+			 */
+			Expr *additional_clause = ts_transform_time_bucket_comparison(constified);
+			if (additional_clause != constified)
```

Comment:
Is it enough to do a pointer comparison here? It seems that the expression could have been changed within but returning the same pointer.

Above you are using the `equal` function so it seems inconsistent.

Reply:
Above I have to use `equal` because it `estimate_expression_value` uses expression mutators which always return a deep copy of the object, even if it's not modified. The time bucket transform is much simpler and returns original object if it's not modified.

Reply:
I made the time bucket transform return NULL when it doesn't apply, to avoid confusion.